backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -443,48 +454,48 @@ public Response createOrUpdate(@PathParam("vaultId") UUID vaultId, @Valid @NotNu
 	@APIResponse(responseCode = "404", description = "no such vault")
 	@APIResponse(responseCode = "409", description = "owned by another user")
 	public Response claimOwnership(@PathParam("vaultId") UUID vaultId, @FormParam("proof") @Valid @ValidJWS String proof) {
-		User currentUser = User.findById(jwt.getSubject());
-		Vault vault = Vault.<Vault>findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
+		User currentUser = userRepo.findById(jwt.getSubject());
+		Vault vault = vaultRepo.findByIdOptional(vaultId).orElseThrow(NotFoundException::new);
 
 		// if vault.authenticationPublicKey no longer exists, this vault has already been claimed by a different user
-		var authPubKey = vault.getAuthenticationPublicKey().orElseThrow(() -> new ClientErrorException(Response.Status.CONFLICT));
+		var authPubKey = vault.getAuthenticationPublicKeyOptional().orElseThrow(() -> new ClientErrorException(Response.Status.CONFLICT));
```

Comment:
a rather unrelated change, isn't it?

note that the `findByIdOptional()` are due to naming conflicts with `findById()`, however in this case the method does not need to carry the utility type in its name. That's like naming a method `HashMap<String, String> getFoobarHashMap()` imo.

Reply:
> unrelated change

No. Every entity uses the jakarta entity persistent property pattern, i.e. accessing the fields via their getter/setter. The getters and setter naming follows the [JavaBeans-style](https://jakarta.ee/learn/docs/jakartaee-tutorial/current/persist/persistence-intro/persistence-intro.html#_persistent_properties). In this case, we have the field `String authenticationPublicKey`. The according getter would be defined as `String getAuthenticationPublicKey()`. But we already have a method with this name with the same erasure.

To additionally getting a parsed key, i renamed the existing getter similar to the Quarkus naming style.